controlplane/kubeadm/internal/kubeadm_config_map.go

```diff
@@ -205,7 +205,6 @@ func (k *kubeadmConfig) UpdateCoreDNSImageInfo(repository, tag string) error {
 		return errors.Wrapf(err, "unable to decode kubeadm ConfigMap's %q to Unstructured object", clusterConfigurationKey)
 	}
 	dnsMap := map[string]string{
-		dnsTypeKey:            string(bootstrapv1.CoreDNS),
```

Comment:
We don't need that as it's default and we don't want to / have to overwrite a pre-existing potentially different value?

Reply:
Shouldn't we still set this as part of the Kubeadm configuration? 

@fabriziopandini

Reply:
We don't need to write it. If this field is empty, kubeadm will default it to CoreDNS;
For existing clusters, if the users are on. something different than CoreDNS, they are also using the SkipCore DNS annotation, so we should be fine

Reply:
Also, https://github.com/kubernetes-sigs/cluster-api/pull/4443 is going to reword how we are managing the Kubeadm config map leverage on API conversions (including this code)